File: backend/BeautyCareClinic.Domain/Entities/Note.cs
namespace BeautyCareClinic.Domain.Entities;

public class Note
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Guid UserId { get; set; }
    public Guid? TreatmentTypeId { get; set; }
    public DateTime NoteDate { get; set; }
    public string Content { get; set; } = string.Empty;
    public string WrittenByFullName { get; set; } = string.Empty;

    public Customer Customer { get; set; } = null!;
    public User User { get; set; } = null!;
    public TreatmentType? TreatmentType { get; set; }
}

File: frontend/src/types/Note.ts
export interface Note {
  id: string;
  customerId: string;
  treatmentTypeId?: string;
  /** Treatment type name — denormalized from API */
  treatmentTypeName?: string;
  /** Content of the note (API field name) */
  content?: string;
  /** Kept as alias for backward compatibility with mock data */
  text?: string;
  /** Server-derived user id of the author */
  userId?: string;
  /** Kept as alias for backward compatibility with mock data */
  authorUserId?: string;
  /** Snapshot of the author's name at writing time */
  writtenByFullName?: string;
  /** ISO date string — note date (API field name) */
  noteDate?: string;
  /** Kept as alias for backward compatibility with mock data */
  createdDate?: string;
}
